src/Campaigns/resources/admin/components/CampaignFormModal/index.tsx

```diff
@@ -0,0 +1,162 @@
+            endDateTime: getDateString(
+                campaign?.endDateTime?.date ? new Date(campaign?.endDateTime?.date) : getNextSharpHour(2)
+            ),
+        },
+    });
+
+    const onSubmit: SubmitHandler<Inputs> = async (inputs) => {
+        try {
+            inputs.startDateTime = getDateString(new Date(inputs.startDateTime));
+            inputs.endDateTime = getDateString(new Date(inputs.endDateTime));
+
+            const endpoint = campaign?.id ? `/campaign/${campaign.id}` : '';
+            const response = await API.fetchWithArgs(endpoint, inputs, 'POST');
+
+            handleClose(response);
+        } catch (error) {
+            console.error('Error submitting campaign campaign', error);
+        }
+    };
+
+    return (
+        <FormModal
+            isOpen={isOpen}
+            handleClose={handleClose}
+            title={title}
+            handleSubmit={handleSubmit(onSubmit)}
+            errors={errors}
+            className={styles.campaignForm}
+        >
+            <div className="givewp-campaigns__form-row">
+                <label htmlFor="title">{__('Campaign Name', 'give')}</label>
+                <input
+                    type="text"
+                    {...register('title', {required: __('The campaign must have a name!', 'give')})}
+                    aria-invalid={errors.title ? 'true' : 'false'}
+                    placeholder={__('Enter campaign name', 'give')}
+                />
+            </div>
+            <div className="givewp-campaigns__form-row">
+                <label htmlFor="shortDescription">{__('Short Description', 'give')}</label>
+                <textarea {...register('shortDescription')} rows={4} />
+            </div>
+            <div className="givewp-campaigns__form-row givewp-campaigns__form-row--half">
+                <div className="givewp-campaigns__form-column">
+                    <label htmlFor="startDateTime">{__('Start date and time', 'give')}</label>
+                    <input
+                        type="datetime-local"
+                        {...register('startDateTime', {required: __('The campaign must have a start date!', 'give')})}
+                        aria-invalid={errors.startDateTime ? 'true' : 'false'}
+                    />
+                </div>
+                <div className="givewp-campaigns__form-column">
+                    <label htmlFor="endDateTime">{__('End date and time', 'give')}</label>
+                    <input type="datetime-local" {...register('endDateTime')} />
+                </div>
+            </div>
+
+            <button
+                type="submit"
+                className={`button button-primary ${!isDirty ? 'disabled' : ''}`}
+                aria-disabled={!isDirty}
+                disabled={!isDirty}
+            >
+                {campaign?.id ? __('Save changes', 'give') : __('Save campaign', 'give')}
+            </button>
+        </FormModal>
+    );
+}
+
+type Campaign = {
```

Comment:
Maybe this is nitpicking, but I would love to see interfaces and types defined at the top of the file.

Reply:
Resolved in 505c618cb7261de08d622eee69c5ff24137b1728